qiskit/wrapper/_wrapper.py

```diff
@@ -136,37 +137,68 @@ def get_backend(name):
-            option.
-        compile_config (dict or None): a dictionary of compile configurations.
-            If `None`, the default compile configuration will be used.
+        circuits (QuantumCircuit or list[QuantumCircuit]): circuits to compile
+        backend (BaseBackend or str): a backend to compile for
+        config (dict): dictionary of parameters (e.g. noise) used by runner
+        basis_gates (str): comma-separated basis gate set to compile to
+        coupling_map (list): coupling map (perhaps custom) to target in mapping
+        initial_layout (list): initial layout of qubits in mapping
+        shots (int): number of repetitions of each circuit, for sampling
+        max_credits (int): maximum credits to use
+        seed (int): random seed for simulators
+        qobj_id (int): identifier for the generated qobj
+        hpc (dict): HPC simulator parameters
         skip_translation (bool): If True, bypass most of the compilation process and
             creates a qobj with minimal check nor translation
     Returns:
         obj: the qobj to be run on the backends
     """
     # pylint: disable=redefined-builtin
-    return qiskit._compiler.compile(list_of_circuits, backend, compile_config, skip_translation)
+    if isinstance(backend, str):
+        backend = _DEFAULT_PROVIDER.get_backend(backend)
+    return qiskit._compiler.compile(circuits, backend,
+                                    config, basis_gates, coupling_map, initial_layout,
+                                    shots, max_credits, seed, qobj_id, hpc,
+                                    skip_translation)
 
 
-def execute(list_of_circuits, backend_name, compile_config=None,
+def execute(circuits, backend,
+            config=None, basis_gates=None, coupling_map=None, initial_layout=None,
+            shots=1024, max_credits=10, seed=None, qobj_id=None, hpc=None,
             skip_translation=False):
     """Executes a set of circuits.
 
     Args:
-        list_of_circuits (list[QuantumCircuits]): list of circuits.
-        backend_name (str): the name of the backend to execute the circuits on.
-        compile_config (dict or None): a dictionary of compile configurations.
+        circuits (QuantumCircuit or list[QuantumCircuit]): circuits to execute
+        backend (BaseBackend or str): a backend to execute the circuits on
+        config (dict): dictionary of parameters (e.g. noise) used by runner
+        basis_gates (str): comma-separated basis gate set to compile to
+        coupling_map (list): coupling map (perhaps custom) to target in mapping
+        initial_layout (list): initial layout of qubits in mapping
+        shots (int): number of repetitions of each circuit, for sampling
+        max_credits (int): maximum credits to use
+        seed (int): random seed for simulators
+        qobj_id (int): identifier for the generated qobj
+        hpc (dict): HPC simulator parameters
```

Comment:
We have to mark this method, and `compile` for refactoring in the future. Too many arguments is usually a symptom that methods have too many responsibilities. Nothing critical, of course...